### PR TITLE
Public js

### PR DESCRIPTION
**Pull request by Nikitak211**
login , register and logout routes were made and checked that working properly and code was cleaned .

### CODE REVIEW

File: controller/AuthControll.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+const User = require('../models/user')
+const bcrypt = require('Bcryptjs')
+const jwt = require('jsonwebtoken')
+require('dotenv').config()
```
Comment (8o8inCodes) [inline controller/AuthControll.js]: This required only in your initial file, once.
No need to require it again elsewhere

File: package.json [hunk 2]
```diff
@@ -17,11 +17,15 @@
...
+    "dotenv": "^10.0.0",
     "express": "^4.17.1",
+    "express-session": "^1.17.2",
+    "jsonwebtoken": "^8.5.1",
     "mongodb": "^4.1.2",
     "mongoose": "^6.0.7",
     "node": "^16.9.1",
-    "nodemon": "^2.0.13",
-    "uuid": "^8.3.2"
+    "nodemon": "^2.0.13"
```
Comment (8o8inCodes) [inline package.json:30]: should be in dev dependencies

File: public/login/login.html [hunk 3]
```diff
@@ -18,7 +18,7 @@
             <li id="notification"><a href="#"></a></li>
         </ul>
     </header>
-    <form class="login_form" id="loginForm" method="GET" action="/">
+    <form class="login_form" id="loginForm" method="POST" action="/">
```
Comment (8o8inCodes) [inline public/login/login.html]: Don't use form's functionality. Do it the same way you did the textsnippet.
Find all the buttons and fields and fetch (remember to preventDefault)

**Review by 8o8inCodes — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -17,11 +17,17 @@
...
+    "dotenv": "^10.0.0",
     "express": "^4.17.1",
+    "express-session": "^1.17.2",
+    "jsonwebtoken": "^8.5.1",
     "mongodb": "^4.1.2",
     "mongoose": "^6.0.7",
-    "node": "^16.9.1",
-    "nodemon": "^2.0.13",
-    "uuid": "^8.3.2"
+    "node": "^16.9.1"
```
Comment (8o8inCodes) [inline package.json]: Node isn't required here

File: public/homepage/logout.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+})
+
+async function logout() {
+
+    const response = await fetch('/api/auth/logout' ,{
+        method: 'POST', // or 'PUT'
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({send:'logout'})
```
Comment (8o8inCodes) [inline public/homepage/logout.js]: Body isn't required here, just if the request enters /logout then just log him out, no to look at the body

File: public/homepage/logout.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    })
+    const json = await response.json()
+    if(req.body.send){
+        res.send({logout: true})
+        location.reload()
+        alert(json.drop)
+    }
```
Comment (8o8inCodes) [inline public/homepage/logout.js]: wtf is this, I think it's a wrong copy paste lol you never defined req/res anywhere

File: public/homepage/script.js [hunk 7]
```diff
@@ -1,6 +1,7 @@
...
+//     bla1: user
+// }
 
-textarea.addEventListener("input", function (e) {
-  this.style.height = "auto";
-  this.style.height = this.scrollHeight + "px";
-});
\ No newline at end of file
+// // this way i can pull the specific data from body.
+// req.body.bal1
```
Comment (8o8inCodes) [inline public/homepage/script.js]: Delete useless comment

File: public/login/login.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(options)
+    })
+    const json = await response.json()
+    if(json.error){
+        errorOn(email, json.login)
+        errorOn(password, json.login)
+        return alert(json.login)
```
Comment (8o8inCodes) [inline public/login/login.js]: Instead of json.login, make it json.message so it will be much clearer.

File: public/login/login.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    } 
+
+    const response = await fetch('/api/auth/login' ,{
+        method: 'POST', // or 'PUT'
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(options)
+    })
+    const json = await response.json()
```
Comment (8o8inCodes) [inline public/login/login.js]: Rename json to "data"

Comment (8o8inCodes) [reply public/login/login.js]: Do it to all the other code

File: public/register/register.js [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+const email = document.getElementById('email');
+const validation_email = document.getElementById('validation_email');
+const register_btn = document.getElementById('register_btn');
+
+document.getElementById("register_form").addEventListener('submit', (event) => {
+	event.preventDefault();
+	checkUser();
+})
+
+async function  checkUser() {
```
Comment (8o8inCodes) [inline public/register/register.js]: Remove extra space

File: public/register/register.js [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+		}
+
+		const response = await fetch('/api/auth/register', {
+			method: 'POST', // or 'PUT'
+			headers: {
+				'Content-Type': 'application/json',
+			},
+			body: JSON.stringify(options),
+		})
+		const json = await response.json()
```
Comment (8o8inCodes) [inline public/register/register.js]: rename json to data

File: public/register/register.js [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+
+		if(json.registerError){
+			errorOn(email, json.registerError)
+			errorOn(validation_email, json.registerError)
```
Comment (8o8inCodes) [inline public/register/register.js]: change to data.message

File: routes/auth.js [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+    try{
+    const {
+        username,
+        password,
+        email
+    } = req.body
+
+    let user = await User.findOne({ email })
+    if (user) {
+        return res.send({ registerError: "email allready exsisting" })
```
Comment (8o8inCodes) [inline routes/auth.js]: "registerError" to "message" and add "error: true"
And then in front end check if error, then do the error logic, do it to all the errors

File: routes/auth.js [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+    user = new User({
+        username,
+        password: hashedPass,
+        email
+    })
+
+    await user.save()
+        res.send({success: "user has created"})
+    } catch (error) {
+        res.send({error: "cannot complete registration"})
```
Comment (8o8inCodes) [inline routes/auth.js]: Same here:
```js
{
   error: true,
   message: "message"
}
```

Keep it consistent like this.

Comment (8o8inCodes) [reply routes/auth.js]: Same with success, but error false

File: routes/auth.js [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+            if (user) {
+                bcrypt.compare(password, user.password, function (err, result) {
+                    if (err) {
+                            return res.send({
+                                loginError: 'password of email doesnt exsist'
+                            })
+                    }
+
+                    if (result) {
+
```
Comment (8o8inCodes) [inline routes/auth.js]: remove empty line

File: routes/auth.js [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+} catch {
+    res.send({
+        login: 'password of email doesnt exsist'
+    })
+}});
+
+router.post('/logout', (req, res) => {
+    if(logout){
+    req.session.destroy((err) => {
+        res.send( {drop: 'logout succsess'})
```
Comment (8o8inCodes) [inline routes/auth.js]: error false
message: "Logged out."

**Review by 8o8inCodes — CHANGES_REQUESTED** [review]

File: public/homepage/logout.js (hunk 6 above)
Comment (Nikitak211) [reply public/homepage/logout.js]: i tried to alert ("succssesfuly logged out")

**Review by 8o8inCodes — APPROVED** [review]

**Pull request merged**